aquia-inc/template-vite-react
# build(deps): update testing-library dependencies to latest

## Summary

Upgraded the `@testing-library` dependencies to their latest versions to maintain compatibility, improve testing reliability, and take advantage of the latest features.

### Added

- None

### Changed

- Upgraded `@testing-library/dom` from `^9.3.4` to `^10.4.0`
- Upgraded `@testing-library/react` from `^14.3.1` to `^16.0.0`

### Deprecated

- None

### Removed

- None

### Fixed

- None

## How to test

1. Checkout this branch: `git checkout origin/build/dependencies/11-testing-library`
2. Run `yarn` and ensure all dependencies are correctly installed without duplication and that there are no issues related to missing or incompatible dependencies.
3. Run `yarn test` and verify that all tests run and pass.
4. Run `yarn dev` and verify that the application runs as expected.
5. Run `yarn build` and verify that the application builds as expected.
6. Run `yarn build-storybook` and verify the Storybook build succeeds.
